src/app/portlets/dot-edit-page/layout/components/dot-theme-selector/dot-theme-selector.component.spec.ts

```diff
@@ -0,0 +1,181 @@
+import { MockDotMessageService } from '../../../../../test/dot-message-service.mock';
+import { Observable } from 'rxjs/Observable';
+import { By } from '@angular/platform-browser';
+import { mockDotThemes } from '../../../../../test/dot-themes.mock';
+import { DataGridModule } from 'primeng/primeng';
+import { SiteSelectorModule } from '../../../../../view/components/_common/site-selector/site-selector.module';
+import { mockSites, SiteServiceMock } from '../../../../../test/site-service.mock';
+import { SiteService } from 'dotcms-js/dotcms-js';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { PaginatorService } from '../../../../../api/services/paginator/paginator.service';
+import { DotThemesServiceMock } from '../../../../../test/dot-themes-service.mock';
+
+describe('DotThemeSelectorComponent', () => {
+    let component: DotThemeSelectorComponent;
+    let fixture: ComponentFixture<DotThemeSelectorComponent>;
+    let de: DebugElement;
+    const messageServiceMock = new MockDotMessageService({
+        'editpage.layout.theme.header': 'Header',
+        'editpage.layout.theme.search': 'Search',
+        'dot.common.apply': 'Apply',
+        'dot.common.cancel': 'Cancel'
+    });
+    const siteServiceMock = new SiteServiceMock();
+    let dialog;
+    let paginatorService: PaginatorService;
+    let dotThemesService: DotThemesService;
+
+    beforeEach(() => {
+        DOTTestBed.configureTestingModule({
+            declarations: [DotThemeSelectorComponent],
+            imports: [DataGridModule, SiteSelectorModule, BrowserAnimationsModule],
+            providers: [
+                {
+                    provide: DotThemesService,
+                    useClass: DotThemesServiceMock
+                },
+                {
+                    provide: DotMessageService,
+                    useValue: messageServiceMock
+                },
+                { provide: SiteService, useValue: siteServiceMock },
+                PaginatorService
+            ]
+        });
+
+        fixture = DOTTestBed.createComponent(DotThemeSelectorComponent);
+        component = fixture.componentInstance;
+        de = fixture.debugElement;
+        dialog = de.query(By.css('p-dialog')).componentInstance;
+        component.value = mockDotThemes[0];
+        paginatorService = de.injector.get(PaginatorService);
+        dotThemesService = de.injector.get(DotThemesService);
+    });
+
+    describe('Dialog', () => {
+        beforeEach(() => {
+            fixture.detectChanges();
+        });
+
+        it('should be visible on init', () => {
+            expect(dialog.visible).toBeTruthy();
+        });
+
+        it('should emit close event when click on cancel button', () => {
+            const cancelBtn = de.query(By.css('.cancel')).nativeElement;
+            spyOn(component.close, 'emit');
+            cancelBtn.click();
```

Comment:
Angular have a trigger event utility, I think we should use it always unless there is something

```
const cancelBtn = de.query(By.css('.cancel'));
cancelBtn.triggerEventHandler('click', {});
```